### ISSUE TITLE
Why don't use exp in crossentropy.py

### ISSUE THREAD

**Issue by thinksoso** [NVIDIA/Megatron-LM#186]
When i debug the program, i find a special code in cross_entropy.py
https://github.com/NVIDIA/Megatron-LM/blob/03d09af09be0f759ad6c9dd60b36e22da67b7c44/megatron/mpu/cross_entropy.py#L76
![image](https://user-images.githubusercontent.com/42956025/151104456-6f91c3d1-6d9f-4e86-a0bf-b7fea3b1a163.png)
and the variables is
![企业微信截图_16431700537122](https://user-images.githubusercontent.com/42956025/151104534-e6f61e16-428a-4233-a32f-350a45e6590a.png)
So, why the predicted_logits is not the **exp** logits but the original logits?

https://colab.research.google.com/drive/1VGnOCd3mMCJh8a4mOfenWSZNDZ1rMopF?usp=sharing
I made a test. Compared with pytorch's crossentropy loss implementation, the operation that don't use exp looks like just scale up the loss and grad. I think it **doesn't hurt** the training process. 
What's the motivation to scale up it? 
Thank you!

( I pulled a same issue before in MicroSoft/DeepSpeed https://github.com/microsoft/DeepSpeedExamples/issues/158
